src/app/index.test.js

```diff
@@ -40,6 +40,179 @@ describe("Templates as a service... again!", () => {
+                href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+                rel="license"
+              >Open Government Licence v3.0</a>, except where otherwise stated
+            </span>
+          </div>
+          <div class="govuk-footer__meta-item">
+            <a
+              class="govuk-footer__link govuk-footer__copyright-logo"
+              href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
+            >© Crown copyright</a>
+          </div>
+        </div>
+      </div>
+    </footer>
+    
+    
+      </body>
+    </html>`
+          return request(app)
+              .post("/govuk/3.5.0/components/govukTemplate")
+              .send({})
+              .expect(200)
+              .then(response => {
+                  expectHtmlToMatch(response.text, expected)
+              })
+      })
+
+      it("should return a customised govukTemplate", () => {
+          const expected = `<!DOCTYPE html>
+<html lang="en" class="govuk-template app-html-class">
+<head>
+    <meta charset="utf-8">
+    <title>GOV.UK - Customised page template</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
+    <meta name="theme-color" content="blue">
+
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+
+    <link rel="shortcut icon" sizes="16x16 32x32 48x48" href="/assets/images/favicon.ico" type="image/x-icon">
+    <link rel="mask-icon" href="/assets/images/govuk-mask-icon.svg" color="blue">
+    <link rel="apple-touch-icon" sizes="180x180" href="/assets/images/govuk-apple-touch-icon-180x180.png">
+    <link rel="apple-touch-icon" sizes="167x167" href="/assets/images/govuk-apple-touch-icon-167x167.png">
+    <link rel="apple-touch-icon" sizes="152x152" href="/assets/images/govuk-apple-touch-icon-152x152.png">
+    <link rel="apple-touch-icon" href="/assets/images/govuk-apple-touch-icon.png">
+
+    <link href="custom-stylesheet.css" rel="stylesheet">
+
+
+    <meta property="og:image" content="/assets/images/govuk-opengraph-image.png">
+</head>
+<body class="govuk-template__body app-body-class">
+<script>document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');</script>
+
+<a href="#main-content" class="govuk-skip-link">Custom skip link</a>
+<header role="banner">Custom header</header>
+<div class="govuk-width-container ">
+    <p>Customised before content, <a class="govuk-link" href="#">this is a link</a>.        <main class="govuk-main-wrapper app-main-class" id="main-content" role="main">
+        <h1 class="govuk-heading-xl">Customised page template</h1>        </main>
+</div>
+
+<footer role="contentinfo">Custom footer</footer>
+<script src="custom-script.js"></script>  </body>
+</html>
+`
+          return request(app)
+              .post("/govuk/3.5.0/components/govukTemplate")
```

Comment:
`govukTemplate` isn't part of `govuk` (a shorthand for `govuk-frontend`), this URL should reflect that.  TBH, I'm wondering whether this should sit in this project or not.  `govuk-frontend` ships without a template.

Reply:
OK, looking at this a bit deeper I think I've overstated it.

I don't think this should be in `/govuk/3.5.0/components` I think this should be `/govuk/3.5.0/template` which means that all the template specific logic can move out of `componentController` into a new `templateController`.